add astro-font tip (#46)
astrolicious/astro-tips.dev: src/content/docs/tips/astro-font.mdx

Reviewer: Can you rename it to `recipes/using-custom-fonts-with-great-dx.mdx`?

---

> title: Setup Astro-Font
> description: Quickly configure Astro-Font for your Astro project
> astroRange: ^4.0.0

Reviewer: ```suggestion
title: Using custom fonts with great DX
description: Learn how to use community projects to use custom fonts easily and with great DX
astroRange: ^4.0.0
```

---

import { Tabs, TabItem, Steps } from "@astrojs/starlight/components";

> 
> Install `astro-font` package using your preferred package manager:

Reviewer: ```suggestion
Using custom fonts is already [officially documented](https://docs.astro.build/en/guides/fonts/) but the process is fairly manual. In this recipe, you'll see how to use community projects to use custom fonts easily and with great DX.

## `astro-font`

Install the `astro-font` package using your preferred package manager:
```

<Tabs>
    <TabItem label="npm">
        ```sh
        npm install astro-font
        ```
    </TabItem>
    <TabItem label="pnpm">
        ```sh
        pnpm add astro-font
        ```
    </TabItem>
    <TabItem label="yarn">
        ```sh
        yarn add astro-font
        ```
    </TabItem>
</Tabs>

> ## With Local Fonts

Reviewer: ```suggestion
### Using local fonts
```

Will be great to show this after the google fonts part

<Steps>
1. Make sure you have the local font in `public/fonts` directory. This example will use IBM Plex Sans
  

> 2. Configure `astro-font` in `layout.astro`

Reviewer: ```suggestion
2. Configure `astro-font` in your layout
```

>    ```astro title="layout.astro"

Reviewer: ```suggestion
   ```astro title="src/layouts/BaseLayout.astro"
```

   ---
   import { AstroFont } from "astro-font";
   ---
   <AstroFont
      config={[

>         {

Reviewer: Shouldn't this be `src`? Haven't checked the docs

Author: src is mentioned couple of lines below. now sure what you meant 😕

          name: "IBMPlexSans",
          src: [
            {
              style: 'normal',
              weight: '400',

>               path: './public/fonts/IBMPlexSans-Regular.ttf'

Reviewer: Highlight this using EC

            },
          ],
          preload: true,
          display: "swap",
          fallback: "sans-serif",
          cssVariable: "font-ibm"
        }
      ]}
    />
   ```

3. Start using the font
  

>    ```css title="globals.css"

Reviewer: ```suggestion
   ```css
```

   body {
       font-family: "font-ibm"
   }
   ```
  
</Steps>

> ## With Google Fonts

Reviewer: ```suggestion
### Using Google Fonts
```

<Steps> 

> 1. Configure `astro-font` in `layout.astro`. This example will use Inter font

Reviewer: ```suggestion
1. Configure `astro-font` in your layout. This example will use Inter
```

>    ```astro title="layout.astro"

Reviewer: ```suggestion
   ```astro title="src/layouts/BaseLayout.astro"
```

   ---
   import { AstroFont } from "astro-font";
   ---
   <AstroFont
      config={[
        {
          src: [],
          name: "Inter",
          preload: false,
          display: "swap",
          fallback: "sans-serif",
          cssVariable: "font-inter",

>           googleFontsURL: 'https://fonts.googleapis.com/css?family=Inter',

Reviewer: Highlight this using EC

        },
      ]}
    />
   ```

2. Start using the font
  

>    ```css title="globals.css"

Reviewer: ```suggestion
   ```css
```

   body {
       font-family: "font-inter"
   }
   ```
  
</Steps>

> ## With TailwindCSS

Reviewer: I think we can just reference the official docs here https://docs.astro.build/en/guides/fonts/#register-fonts-in-tailwind

   ```astro title="layout.astro"
   ---
   import { AstroFont } from "astro-font";
   ---
   <AstroFont
      config={[
        {
          name: "Eduns",
          src: [
            {
              style: 'normal',
              path: 'https://fonts.gstatic.com/s/edunswactfoundation/v2/raxRHjqJtsNBFUi8WO0vUBgc9D-2lV_oQdCAYlt_QTQ0vUxJki9tosGMci4.woff2'
            },
          ],
          preload: true,
          display: "swap",
          fallback: "sans-serif",
          cssVariable: "font-eduns",
        },
      ]}
    />
   ```

2. Update tailwind config file
  
   ```js title="tailwind.config.mjs" ins={1,5-7}
    import defaultTheme from 'tailwindcss/defaultTheme';
    export default {
      theme: {
        extend: {
          fontFamily: {
            sans: ["var(--font-eduns)", ...defaultTheme.fontFamily.sans],
          }
        },
      },
      plugins: [],
    }
   ```

3. Start using the font
  
   ```astro title="index.astro" ins="font-sans"
      <p class="bg-blue-950 font-sans">
			Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla tempus lacinia luctus.
      </p>
   ```
  
</Steps>

:::note

> Make sure to read through [astro-font README](https://github.com/rishi-raj-jain/astro-font/blob/master/packages/astro-font/README.md) incase your deployment service does not supports Node.js runtime

Reviewer: ```suggestion
Make sure to read through [astro-font README](https://github.com/rishi-raj-jain/astro-font/blob/master/packages/astro-font/README.md) in case your deployment service does not supports Node.js runtime
```
